tests/conftest.py

```diff
@@ -1,19 +1,144 @@
 import pytest
+import sys
+import json
+import logging
+from pathlib import Path
+from dotenv import load_dotenv
+from typing import Generator, Any
 
+# DSI: Add project root to path for script imports
+project_root = Path(__file__).parent.parent
+sys.path.insert(0, str(project_root))
```

Comment:
I'm not sure this is a good idea. Messing around with the path has caused issues in the past. And the supported method of running this should me using the makefile, in which case the user's current dir is known.